reference/src/transforms/window.md

```diff
@@ -0,0 +1,56 @@
+# Window
+
+Applies a pipeline to overlapping segments of rows.
+
+```prql_no_test
+window rows:.. range:.. expanding:false rolling:0 {pipeline}
+```
+
+For each row in result, its input segment is determined either by:
+
+- `rows`, which takes a range of rows relative to current row,
+- `range`, which takes a range of values relative to current row value.
+
+Start of the range is inclusive, end of range is exclusive. Index 0 references
+current row. If a bound is omitted, segment will extend until the end of the table (or group).
+
+For example:
+- `rows:0..3`   means current row plus two following,
```

Comment:
I was thinking how to do `current` — this approach is perfect, can't believe I didn't think of it

Reply:
Though I'm still thinking through the answer to the question "what number is the current row?".

Did you think about making `0` the current row on both sides, and the range being inclusive?

It looks like that's how it's defined here:

```
`rows:-3..0`  means two preceding rows plus current row,
```

but possibly not here:

```
`rows:..1`    means all rows from the start of the table to and including current row,
```

Or am I missing something?

Reply:
Oh, that's a mistake. It should be 
```
`rows:-2..1` means two preceding rows plus current row,
```

Such indexing was supported [by Dijkstra](https://www.cs.utexas.edu/users/EWD/transcriptions/EWD08xx/EWD831.html) it it seem most natural to me. I also wanted this to be similar to python and rust, while distancing from matlab...

Reply:
Wonderful reference and explanation.

Do you think we should have a slightly different syntax for this, like rust has, then? My concern is that SQL ranges are by default inclusive; e.g. `BETWEEN 0 and 3`, or `in 0..3` includes both 0 & 3, as does `range:0..3`

There are a few areas like this where it's difficult to completely upgrade SQL, because we can't see through to the data. So we can either separate off the parts we can — e.g. different approach to `rows` than other ranges — or compromises to their standards — e.g. just use inclusive for everything.

WDYT?